tests/CredentialRequestClient.spec.ts

```diff
@@ -0,0 +1,159 @@
+import { KeyObject } from 'crypto';
+
+import * as jose from 'jose';
+import { KeyLike, VerifyOptions } from 'jose/dist/types/types';
+import nock from 'nock';
+import * as u8a from 'uint8arrays';
+
+import {
+  createProofOfPossession,
+  CredentialRequest,
+  CredentialRequestClient,
+  CredentialResponse,
+  ErrorResponse,
+  JWS_NOT_VALID,
+  JWTSignerArgs,
+  ProofOfPossession,
+} from '../lib';
+
+const partialJWT =
+  'eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6ImRpZDpleGFtcGxlOmViZmViMWY3MTJlYmM2ZjFjMjc2ZTEyZWMyMS9rZXlzLzEifQ.eyJhdWQiOiJodHRwczovL29pZGM0dmNpLmRlbW8uc3BydWNlaWQuY29tL2NyZWRlbnRpYWwiLCJp';
+
+// Must be JWS
+const signJWT = async (args: JWTSignerArgs): Promise<string> => {
+  const { header, payload, privateKey } = args;
+  return await new jose.CompactSign(u8a.fromString(JSON.stringify({ ...payload })))
+    .setProtectedHeader({ ...header, alg: args.header.alg })
+    .sign(privateKey);
+};
+
+const verifyJWT = async (args: { jws: string | Uint8Array; key: KeyLike | Uint8Array; options?: VerifyOptions }): Promise<void> => {
+  await jose.compactVerify(args.jws, args.key, args.options);
+};
+
+const jwtArgs: JWTSignerArgs = {
+  header: {
+    alg: 'ES256',
+    kid: 'did:example:ebfeb1f712ebc6f1c276e12ec21/keys/1',
+  },
+  payload: {
+    iss: 's6BhdRkqt3',
+    nonce: 'tZignsnFbp',
+    jti: 'tZignsnFbp223',
+  },
+  privateKey: undefined,
+  publicKey: undefined,
+};
+
+beforeAll(async () => {
+  const keyPair = await jose.generateKeyPair('ES256');
+  jwtArgs.privateKey = keyPair.privateKey as KeyObject;
+  jwtArgs.publicKey = keyPair.publicKey as KeyObject;
+});
+
+describe('VcIssuanceClient ', () => {
+  it('should build correctly provided with correct params', function () {
+    const vcIssuanceClient = CredentialRequestClient.builder()
```

Comment:
Perhaps a comprehensice replace All needs to run .

vcIssuanceClient needs to be renamed.

Reply:
Renamed